### ISSUE TITLE
Dragging markers

### ISSUE THREAD

**Issue by Shemeikka** [Leaflet/Leaflet.markercluster#685]
Hi,

I'm having trouble dragging markers on map. I have read old issues regarding this topic and if I have understood correctly, one needs to first remove marker from cluster on dragstart-event and add it back to cluster on dragend-event. Also this marker needs to be added to the map after removing it from cluster so that it doesn't disappear when moving it.

```
var markers = L.markerClusterGroup({ disableClusteringAtZoom: 12 });

var marker = L.marker([pos.lat, pos.lng], { title: "title", draggable: "true" });
marker.on("dragstart", function (ev) {            
    console.log("drag start");
    markers.removeLayer(this);
    marker.addTo(mMapView);            
});
marker.on("drag", function (ev) {
    console.log("drag");
    this.setLatLng(ev.latlng);
});
marker.on("dragend", function (ev) {
    console.log("drag end");
    marker.removeFrom(mMapView);
    markers.addLayer(this);
});

markers.addLayer(marker); 
```

This almost works. When I try to drag the marker, it doesn't move at all (dragstart-event fires but not others). After the mouse button is released and I try dragging again, it works.

Any ideas on how I could get this to work?

I'm using:
Leaflet v1.0.0-rc.1
Leaflet-markercluster v1.0.0-rc.1
Latest Chrome on Windows 8.1

**Comment by danzel** [comment]
On 1.0.0 you shouldn't need to do anything AFAIK. just use dragging support and it should be fine.

**Comment by julienVoltzenlogel** [comment]
I am in leaflet v1.0.0 and MarkerCluster 1.0.0.

I have a bug with MarkerCluster :
When I start dragging à marker, the hand icon off the browser is closed and I can't drop my marker.

If I add a marker without using the MarkerCluster plugin, the drag and drop work perfectly.

```
drawnMarkers = new L.MarkerClusterGroup({
		maxClusterRadius: 100,
		chunkedLoading: true,
		showCoverageOnHover: false,
		removeOutsideVisibleBounds: true,
		zoomToBoundsOnClick: true,
		singleMarkerMode: false
	});
	
	var compteur = 1;

	markers.forEach(function (jsonData) {
		var marker = L.marker([jsonData.Latitude, jsonData.Longitude], { draggable: true });

		drawnMarkers.addLayer(marker);

		// On recentre l'affichage sur les markers quand ils ont tous été ajouté sur la carte
		if (compteur == markers.length) {
			var group = new L.featureGroup(drawnMarkers.getLayers());
			map.fitBounds(group.getBounds());
		}		

		compteur++;
	});

	map.addLayer(drawnMarkers);
```

**Comment by htejwani** [comment]
Hi,

Any update on this? I am facing the same issue with MarkerCluster plugin. without it, the dragging works fine.

**Comment by Crapal** [comment]
Same problem here :/

**Comment by ghybs** [comment]
Hi,

Might be related to https://github.com/Leaflet/Leaflet/issues/5063
But not sure that is enough for this issue.

**Comment by Kae2312** [comment]
Same problem here, is it possible to drag marker from MarkerClusterGroup?

**Comment by hh-h** [comment]
Hi,

Thanks for awesome work with this plugin!

Any update on this?
Strange, but if you drag a marker quickly, sometimes it can move a few millimeters :)

**Comment by danzel** [comment]
I've taken a look at this and yeah it's broken, will try figure it out.
http://playground-leaflet.rhcloud.com/tuja/1/edit?html,output

**Comment by danzel** [comment]
I've pushed up a fix for this. You'll need to build markercluster, or grab a temporary build from here:
https://gist.github.com/danzel/77e3b3372ce751c8a825e55473df2b51#file-leaflet-markercluster-src-js

Please test this out and report back, will push a new build if I get a positive report or two :+1:

**Comment by hh-h** [comment]
yay! Markers are draggable now :no_mouth: 

Thanks, @danzel

**Comment by ryanjhodge** [comment]
I built from the source and tested the `marker-clustering-dragging.html` example and it's working as expected.

**Comment by danzel** [comment]
1.0.1 tagged and npm'd :+1:

**Comment by JuH33** [comment]
Hello,
i'm actually facing the same issue. I tried your fix sent above. However i'm wondering if your fix has been patched in a further version of Leaflet Markercluster. Can you please confirm @danzel ?
Thanks =)

**Comment by hh-h** [comment]
@JuH33 You should provide version you are using.

**Comment by JuH33** [comment]
Oops, thanks @hh-h  ! 
1.0.4 =)

**Comment by danzel** [comment]
The fix is definitely in 1.0.4.
If you can make a small reproducable case, please open a new bug with it.
This works as far as I'm aware :)

**Comment by ghybs** [comment]
Hi,

Works with Leaflet 1.0.3 + Leaflet.markercluster 1.0.4:
https://jsfiddle.net/3v7hd2vx/245/
